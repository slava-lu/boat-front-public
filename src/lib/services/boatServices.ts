import { BASE_URL } from "@/lib/const";
import { Boat, BoatValues } from "@/lib/types";

const boatsUrl = `${BASE_URL}/boats`;

export async function getBoats(): Promise<Boat[]> {
  try {
    const response = await fetch(boatsUrl, {
      next: { cache: "no-store" },
    } as RequestInit & { next: { cache: string } });

    if (!response.ok) {
      console.error(`Failed to fetch boats: ${response.status}`);
    }

    return response.json();
  } catch (error) {
    console.error("Error fetching boats data:", error);
    return [];
  }
}

export async function getBoat(id: string): Promise<Boat> {
  try {
    const response = await fetch(`${boatsUrl}/${id}`, {
      next: { cache: "no-store" },
    } as RequestInit & { next: { cache: string } });

    if (!response.ok) {
      console.error(`Failed to fetch boat: ${response.status}`);
    }

    return response.json();
  } catch (error) {
    console.error("Error fetching boat data:", error);
    throw error;
  }
}

export async function deleteBoat(id: string) {
  try {
    const response = await fetch(`${boatsUrl}/${id}`, {
      method: "DELETE",
    });

    if (!response.ok) {
      console.error(`Failed to delete boat: ${response.status}`);
    }
    return response.ok;
  } catch (error) {
    console.error("Error deleting boat:", error);
    return false;
  }
}

export async function addBoat(data: BoatValues): Promise<boolean> {
  try {
    const response = await fetch(boatsUrl, {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify(data),
    });

    if (!response.ok) {
      console.error(`Failed to update boat: ${response.status}`);
      return false;
    }
    return true;
  } catch (error) {
    console.error("Error updating boat:", error);
    return false;
  }
}

export async function updateBoat(
  id: string,
  data: BoatValues,
): Promise<boolean> {
  try {
    const response = await fetch(`${boatsUrl}/${id}`, {
      method: "PUT",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify(data),
    });

    if (!response.ok) {
      console.error(`Failed to update boat: ${response.status}`);
      return false;
    }
    return true;
  } catch (error) {
    console.error("Error updating boat:", error);
    return false;
  }
}
