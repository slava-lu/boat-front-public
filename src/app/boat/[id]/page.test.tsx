import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import BoatDetailPage from "./page";
import { getBoat } from "@/lib/services/boatServices";

jest.mock("@/lib/services/boatServices");
jest.mock("next/navigation", () => ({
  useRouter: jest.fn(() => ({
    push: jest.fn(),
    refresh: jest.fn(),
  })),
}));

describe("BoatDetailPage", () => {
  it('should render "Boat details not found." if boat data is null', async () => {
    (getBoat as jest.Mock).mockResolvedValue(null);
    const { findByText, asFragment } = render(
      <BoatDetailPage params={{ id: "123" }} />,
    );

    expect(await findByText("Boat details not found.")).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
