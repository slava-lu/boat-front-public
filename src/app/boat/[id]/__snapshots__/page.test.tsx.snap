// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoatDetailPage should render "Boat details not found." if boat data is null 1`] = `
<DocumentFragment>
  <div>
    Boat details not found.
  </div>
</DocumentFragment>
`;
